src/app/comments/comment-form-container/comment-form-container.component.ts

```diff
@@ -12,38 +12,38 @@ import {switchMap, tap} from "rxjs/operators";
 })
 export class CommentFormContainerComponent implements OnInit {
 
-  ngOnInit(): void {
-  }
 
-  constructor(private route: ActivatedRoute, private commentService: CommentService ){
-  }
 
   errorMessage:string = '';
-  successfulSending:boolean;
-
-  comment: NewComment = {
-    body: null,
-    postId: null,
-    firstName: null,
-    lastName: null,
-    email: null,
+  isSendSuccessful:boolean;
+  @Input() postId: number;
+  @Output() response = new EventEmitter<any>();
+
+
+  constructor(private route: ActivatedRoute, private commentService: CommentService ){
   }
 
-  getCommentDataFromParent(data){
-    this.comment.firstName = data.firstName;
-    this.comment.lastName = data.lastName;
-    this.comment.email = data.email;
-    this.comment.body = data.body;
+  newComment(data: NewComment){
+    let comment: NewComment = {
+      firstName: data.firstName,
+      lastName: data.lastName,
+      email: data.email,
+      body: data.body,
+      postId: this.postId,
+      acceptedTerms: data.acceptedTerms
```

Comment:
the terms & conditions checkbox is only for validation, it has no value for us to keep it in db. Suggest sending the data without this data.